extensions/reactive-routes/runtime/src/main/java/io/quarkus/vertx/web/runtime/VertxWebRecorder.java

```diff
@@ -146,12 +146,7 @@ public void handle(RoutingContext event) {
                         @Override
                         protected void proceed(Throwable throwable) {
 
-                            if (event.failed()) {
-                                //auth failure handler should never get called from route failure handlers
-                                //but if we get to this point bad things have happened,
-                                //so it is better to send a response than to hang
-                                event.end();
```

Comment:
@michalvavrik One thing which is worth checking why this code was added, Stuart may've added it to avoid a concrete case where the request was hanging

Reply:
@sberyozkin I've added that code as I was pretty scared someone will use default auth failure handler differently to what we do know and HTTP request will hang (never respond). I didn't know in that point event can already be failed while not called from failure handler. I have one argument why it's safe to remove it (apart of that if we end it, failure handler will never actually receive the event in this particular scenario) and that is look how the default auth failure is called (PR descritpion), if the handler is null, nothing happens already.

I can summarize is even shorter: I was wrong previously, sorry.

Reply:
@michalvavrik Oh sorry, if it was you had added it :-)